Department Heads joining late or at roundstart will have their department underlings notified

Chiefwaffles opened the pull request:
**This is both my first pull request and my actual first foray into byond coding!**
So expect (and please point out) horrible things.

The system now works in a way very similar to the AI late-join announcements. When a new head joins, the AI will announce to the respective department radios: "Person McScience, Research Director, is the new department head"

This also works for things like HoP who have control over two departments - the AI will announce his presence to both channels.

Two main flaws with this approach:
- Can get a bit spammy for the AI at roundstart
- Doesn't work without an AI

Heads that will notify their underlings:
- Head of Security
- Research Director
- Head of Personnel (Notifies all of Cargo and Citizen jobs)
- Chief Medical Officer
- Chief Engineer

Commit:
Underlings are now told who their head is

When heads are present at roundstart, or latejoin the round, all their
alive underlings will be notified.

Incoming5643 commented on code/controllers/subsystem/jobs.dm:
```diff
@@ -390,6 +390,7 @@ var/datum/subsystem/job/SSjob
 		else //We ran out of spare locker spawns!
 			break
 
+
```
if you're commiting with something like tortoisegit you should always open the code on the commit window to look for these empty line only instances, you can safely not commit this file without it affecting anything. Of course now that it's been commited it's a little harder to remove in a way that vanishes it from this PR. So don't worry about it for now.

WJohn commented:
It should probably do the big red notification like with a latejoining captain, it seems weird to get a usually personal blue message used instead, like you just innately *know* who your department is now.

Chiefwaffles commented:
This better?
![ss13 screen](https://cloud.githubusercontent.com/assets/4297806/8560517/63ef4f20-24cd-11e5-889d-ac490cdcf3d7.png)

WJohn commented:
That's fine for roundstart, but I mean for latejoining. It should do the big announcement like with captains "captain joe on deck!" kind of deal. "Research Director Schmoe on deck!"

PKPenguin321 commented:
maybe leave that for another PR, this looks fine for now with the red text at roundstart I think

Commit:
Made message 'boldannounce' and added rank before name

The span class of the message is now "boldannounce", and the rank of
your head is given before their name.

Chiefwaffles commented:
Well, as of now you get that red message when your head joins, or at roundstart. I also added the rank before the name, so people like Cargo/Civilians don't get too confused. It looks a bit like:

"Head of Personnel Francis Bode is your department head!"

SconesC commented:
You could make some automated thingy say it over the channel

Or just give heads a CAPTAIN ON DECK-esque announcement

Chiefwaffles commented:
I'd rather not over-modify each job's procs to check for latejoining, so I can probably do one of these three things:
* I can keep it how it is now
* I can make it like the Captain with a priority announcement 100% of the time (leaning towards this)
* or I can (try to) do the automated radio message thing.

Jalleo commented:
Do the same as the captain for consistency

Chiefwaffles commented:
Well, I take it back. I forgot that you can easily replicate the announcement stuff without actually doing any real copy+pasting. I haven't committed it yet because I'm not sure if it fits. (Announcement type thing clashes with how it's a "personal" kind of message, and it can seem weird on roundstart; especially with a captain also there.)

This it'd look on roundstart:
![ss13 screen2](https://cloud.githubusercontent.com/assets/4297806/8584079/d7f80046-2587-11e5-8fb3-e28f1a354e62.png)

Keep in mind that this will appear with the Captain's announcement if there is also one on roundstart. But otherwise, the message will appear like this whenever you get a department head, and still only for the underlings.

Commit:
Radically changes branch to work similar to AI late-join announcements in that the AI broadcasts the message to the respective department.

Chiefwaffles commented:
Okay, I actually did the final change. I don't think it necessitates a screenshot due to how similar it is to the arrivals announcement system, but I can provide one if needed.

The system now works in a way very similar to the AI late-join announcements. When a new head joins, the AI will announce to the respective department radios: "Person McScience, Research Director, is the new department head"

This also works for things like HoP who have control over two departments - the AI will announce his presence to both channels.

Two main flaws with this approach:
* Can get a bit spammy for the AI at roundstart
* Doesn't work without an AI

Chiefwaffles committed:
Update Chiefwaffles-Who_is_my_boss.yml

Made changelog a bit more accurate.

Dorsisdwarf commented:
I don't like the new AI notification system. Sorry.

Chiefwaffles commented:
Could you elaborate why?

Chiefwaffles commented:
Closing because I made changes I feel are substantial enough for a new PR.
